libgadget/slotsmanager.c

```diff
@@ -379,8 +379,15 @@ slots_gc_sorted()
         PartManager->NumPart--;
     }
 
-    /*Set up ReverseLink*/
-    slots_gc_mark();
+    int gc_slots = 0;
+    for(ptype = 0; ptype < 6; ptype ++)
+        if(SLOTS_ENABLED(ptype))
+            gc_slots = 1;
+
+    if(gc_slots) {
+        /*Set up ReverseLink*/
+        slots_gc_mark();
+    }
 
     for(ptype = 0; ptype < 6; ptype++) {
         if(!SLOTS_ENABLED(ptype))
```

Comment:
Also feels like the sorting and reduce can be moved into gc_slots_collect.

Reply:
No, because it isn't done in slots_gc_slots (the non-sorting gc). In that case the slots array is already sorted in the same order as the particles.